src/lib/y2packager/repomd_downloader.rb

```diff
@@ -0,0 +1,66 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2019 SUSE LLC, All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+# ------------------------------------------------------------------------------
+
+require "rexml/document"
+require "yast"
+
+Yast.import "Pkg"
+
+module Y2Packager
+  # This class downloads the primary.xml metadata file from an rpm-md repository.
+  # For multi-product repositories it downloads the files from all product
+  # subdirectories.
+  class RepomdDownloader
+    attr_reader :url, :product_repos
```

Comment:
what about having product_repos lazy loaded?
like
```
def product_repos
  @product_repos if @product_repos
  
  download
  @product_repos
end
```
and having download private method?